src/test/java/com/aws/greengrass/clientdevices/auth/iot/ThingTest.java

```diff
@@ -126,4 +134,20 @@ void testEquals() {
         assertThat(Thing_CertA, not(equalTo(Thing_CertB)));
         assertThat(Thing_MultiCert, equalTo(Thing_MultiCert_copy));
     }
+
```

Comment:
We are missing a few tests here. I am ok with doing them later but testing that you attachSomething and then set a ttl and then wait for the time to pass and ask for it again and it should no longer be active. See how we are doing it here https://github.com/aws-greengrass/aws-greengrass-client-device-auth/blob/main/src/integrationtests/java/com/aws/greengrass/integrationtests/certificateauthority/OfflineAuthenticationTest.java#L120

Reply:
will have these tests in a separate PR.